Source/WebKit/WPEPlatform/wpe/wayland/WPEToplevelWayland.cpp

```diff
@@ -567,8 +570,6 @@ static void wpeToplevelWaylandConstructed(GObject *object)
         zwp_linux_dmabuf_feedback_v1_add_listener(priv->dmabufFeedback, &linuxDMABufFeedbackListener, object);
     }
 
-    wl_display_roundtrip(wpe_display_wayland_get_wl_display(display));
-
```

Comment:
Why do you need to move this? I think this was needed here before trying to get the current monitor.

Reply:
I needed to change because doing the roundtrip was actually triggering getting the display information in `ScreenManagerWPE::collectScreenProperties()`, and changing the monitor data in a second `configure_bounds` was not re-updating the collected data.

Maybe when changing monitor properties `ScreenManager` needs to collect the data again? (Currently it listens only to display's `monitor-added`/`monitor-removed` in `ScreenManager`'s constructor for updating)

Reply:
I'm still worried about this change, I think it can break setting the initial device scale factor.

Reply:
I've checked this is not a problem.